### PR TITLE
Pass 3 over testing guide

### PR DESCRIPTION
**Pull request by vipulnsward**
- Various grammar fixes
- Added assertions for update controller action tests
- Added user helper tests
- Fix typos

### CODE REVIEW

**Comment by maclover7** [comment]
r? @maclover7

Will take a look at this and review later today.

File: guides/source/testing.md [hunk 1]
```diff
@@ -650,8 +650,7 @@ You should test for things such as:
 * was the correct object stored in the response template?
 * was the appropriate message displayed to the user in the view?
 
-The easiest way to see functional tests in action is to generate a controller
-scaffold:
+The easiest way to see functional tests in action is to generate a controller using scaffold:
```
Comment (maclover7) [inline guides/source/testing.md]: using the scaffold generator

File: guides/source/testing.md [hunk 2]
```diff
@@ -677,7 +676,7 @@ create test/controllers/articles_controller_test.rb
 ...
 ```
 
-Let me take you through one such test, `test_should_get_index` from the file `articles_controller_test.rb`.
+Lets take a look at one such test, `test_should_get_index` from the file `articles_controller_test.rb`.
```
Comment (maclover7) [inline guides/source/testing.md]: Let's

File: guides/source/testing.md [hunk 3]
```diff
@@ -705,7 +704,7 @@ The `get` method kicks off the web request and populates the results into the re
 
 * `flash`: option with a hash of flash values.
 
-All the keyword arguments are optional.
+All these keyword arguments are optional.
```
Comment (maclover7) [inline guides/source/testing.md]: All of these

File: guides/source/testing.md [hunk 4]
```diff
@@ -913,14 +912,19 @@ We can also add a test for updating an existing Article.
...
+  # Reload association to fetch updated data and assert that title is updated. 
+  article.reload
+  assert_equal "updated", article.title 
 end
 ```
 
 Notice we're starting to see some duplication in these three tests, they both access the same Article fixture data. We can D.R.Y. this up by using the `setup` and `teardown` methods provided by `ActiveSupport::Callbacks`.
 
-Our test should now look something like this, disregard the other tests we're leaving them out for brevity.
+Our test should now look something like follows. Disregard the other tests for now, we're leaving them out for brevity.
```
Comment (maclover7) [inline guides/source/testing.md]: like what follows

File: guides/source/testing.md [hunk 5]
```diff
@@ -1123,7 +1132,7 @@ In order to test that your mailer is working as expected, you can use unit tests
 
 For the purposes of unit testing a mailer, fixtures are used to provide an example of how the output _should_ look. Because these are example emails, and not Active Record data like the other fixtures, they are kept in their own subdirectory apart from the other fixtures. The name of the directory within `test/fixtures` directly corresponds to the name of the mailer. So, for a mailer named `UserMailer`, the fixtures should reside in `test/fixtures/user_mailer` directory.
 
-When you generated your mailer, the generator creates stub fixtures for each of the mailers actions. If you didn't use the generator you'll have to make those files yourself.
+When you generated your mailer, the generator creates stub fixtures for each of the mailers actions. If you didn't use the generator you'll have to create those files yourself.
```
Comment (maclover7) [inline guides/source/testing.md]: If you didn't use the generator,

^ just add a comma

File: guides/source/testing.md [hunk 6]
```diff
@@ -1204,7 +1213,7 @@ Testing Jobs
 ------------
 
 Since your custom jobs can be queued at different levels inside your application,
-you'll need to test both jobs themselves (their behavior when they get enqueued)
+you'll need to test both, jobs themselves (their behavior when they get enqueued)
```
Comment (maclover7) [inline guides/source/testing.md]: the jobs themselves

File: guides/source/testing.md [hunk 7]
```diff
@@ -1255,7 +1264,7 @@ end
 Testing Time-Dependent Code
 ---------------------------
 
-Rails provides inbuilt helper methods that enable you to assert that your time-sensitve code works as expected.
+Rails provides inbuilt helper methods that enable you to assert that your time-sensitive code works as expected.
```
Comment (maclover7) [inline guides/source/testing.md]: built-in

**Comment by maclover7** [comment]
A few comments, and then ready to merge!

File: guides/source/testing.md (hunk 7 above)
Comment (vipulnsward) [inline guides/source/testing.md]: http://ell.stackexchange.com/questions/16886/built-in-in-built-inbuilt-and-builtin-im-confused
http://www.merriam-webster.com/interstitial-ad?next=/dictionary/built-in

Going to leave as is, as they are mostly interchangeable and the current one is more suitable in context of programs.

Comment (maclover7) [inline guides/source/testing.md]: IMO, "inbuilt" is not proper grammar, and "built-in"/"builtin" is the way to go. This is my first time hearing someone using the word "inbuilt". :(

**Comment by maclover7** [comment]
??

**Comment by maclover7** [comment]
@vipulnsward why did you merge? I wasn't done reviewing yet :(

**Pull request merged**